modules/phase_field/src/action/RigidBodyMultiKernelAction.C

```diff
@@ -0,0 +1,133 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RigidBodyMultiKernelAction.h"
+#include "Factory.h"
+#include "Parser.h"
+#include "Conversion.h"
+#include "FEProblem.h"
+
+template<>
+InputParameters validParams<RigidBodyMultiKernelAction>()
+{
+  InputParameters params = validParams<Action>();
+
+  params.addRequiredParam<unsigned int>("op_num", "specifies the number of grains to create");
+  params.addRequiredParam<std::string>("var_name_base", "specifies the base name of the variables");
+  params.addParam<bool>("implicit", true, "Whether kernels are implicit or not");
+  params.addParam<bool>("use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
+  params.addParam<std::string>("kappa_name", "kappa_op", "The kappa used with the kernel");
+  params.addParam<std::string>("mob_name", "L", "The mobility used with the kernel");
+  params.addParam<std::string>("f_name", "Base name of the free energy function F defined in a DerivativeParsedMaterial");
+  params.addParam<VariableName>("c", "Name of coupled concentration variable");
+
+  return params;
+}
+
+RigidBodyMultiKernelAction::RigidBodyMultiKernelAction(const InputParameters & params) :
```

Comment:
btw. the supermajority of constructors in Moose calls this `parameters` and _not_ `params`